packages/@webex/plugin-cc/src/services/core/aqm-reqs.ts

```diff
@@ -0,0 +1,314 @@
+          if (isStopped) {
+            reject(new Err.Details('Service.aqm.reqs.TimeoutEvent', {key: k}));
+          }
+        }, ms);
+      });
+    };
+
+    return evt;
+  }
+
+  private async makeAPIRequest<TRes, TErr>(c: Req<TRes, TErr>, cbRes?: CbRes<TRes>): Promise<TRes> {
+    return this.createPromise(c, cbRes);
+  }
+
+  private createPromise<TRes, TErr>(c: Req<TRes, TErr>, cbRes?: CbRes<TRes>) {
+    return new Promise<TRes>((resolve, reject) => {
+      const keySuccess = this.bindPrint(c.notifSuccess.bind);
+      const keyFail = c.notifFail ? this.bindPrint(c.notifFail.bind) : null;
+      const keyCancel = c.notifCancel?.bind ? this.bindPrint(c.notifCancel.bind) : null;
+      let k = '';
+      if (this.pendingRequests[keySuccess]) {
+        k = keySuccess;
+      }
+      if (keyFail && this.pendingRequests[keyFail]) {
+        k += keyFail;
+      }
+      if (k && c.timeout !== 'disabled') {
+        reject(
+          new Err.Details('Service.aqm.reqs.Pending', {
+            key: k,
+            msg: 'The request has been already created, multiple requests are not allowed.',
+          })
+        );
+
+        return;
+      }
+
+      let isClear = false;
+      const clear = () => {
+        delete this.pendingRequests[keySuccess];
+        if (keyFail) {
+          delete this.pendingRequests[keyFail];
+        }
+        if (keyCancel) {
+          delete this.pendingNotifCancelrequest[keyCancel];
+        }
+        isClear = true;
+      };
+
+      this.pendingRequests[keySuccess] = {
+        check: (msg: Msg) => this.bindCheck(c.notifSuccess.bind, msg),
+        handle: (msg: Msg) => {
+          clear();
+          resolve(msg as any);
+        },
+      };
+      if (keyCancel) {
+        this.pendingRequests[keySuccess].alternateBind = keyCancel;
+        this.pendingNotifCancelrequest[keyCancel] = {
+          check: (msg: Msg) => this.bindCheck(c.notifCancel?.bind, msg),
+          handle: (msg: Msg) => {
+            const alternateBindKey = this.pendingNotifCancelrequest[keyCancel].alternateBind;
+            if (alternateBindKey) {
+              this.pendingRequests[alternateBindKey].handle(msg);
+            }
+          },
+          alternateBind: keySuccess,
+        };
+      }
+
+      if (keyFail) {
+        this.pendingRequests[keyFail] = {
+          check: (msg: Msg) => this.bindCheck(c.notifFail!.bind, msg),
+          handle: (msg: Msg) => {
+            clear();
+            const notifFail = c.notifFail!;
+            if ('errId' in notifFail) {
+              LoggerProxy.logger.log(`Routing request failed: ${msg}`);
+              const eerr = new Err.Details(notifFail.errId, msg as any);
+              LoggerProxy.logger.log(`Routing request failed: ${eerr}`);
+              reject(eerr);
+            } else {
+              reject(notifFail.err(msg as any));
+            }
+          },
+        };
+      }
+      let resAxios: WebexRequestPayload | null = null;
```

Comment:
nitpick: We are not using axios so let's avoid that in the names

Reply:
addressed